### ISSUE TITLE
Slow local commit since update to 1.89.2.0

### ISSUE THREAD

**Issue by RauknRolf** [ForkIssues/TrackerWin#2023]
Since I updated to version 1.89.2.0 I have experienced very slow commits. It is possible that it occurred already in version 1.89.1 but I'm not sure... It now takes up to 30 seconds to perform a commit. I have changed my preferences to use my local installation of git (2.31.1.windows.1) instead of the Fork git instance (2.42.0.windows.2) but without any improvement. Any ideas?

**Comment by DanPristupov** [comment]
Open Windows Task Manager and check what process uses most CPU when you commit. Most likely it's antivirus. Consider to add the source directory to the white list.

**Comment by RauknRolf** [comment]
Thanks for the very quick response. I can't see that it is my antivirus. Even if it is, I unfortunately do not have the administrative rights to add to the whitelist. Attaching Task Manager screenshot from my last commit.
![image](https://github.com/fork-dev/TrackerWin/assets/145426213/872796fd-d9e8-4133-b9df-35a51cb622e8)

**Comment by DanPristupov** [comment]
OK. 
1. Can you reproduce the problem in any repo?
2. Do you have a commit hook? Check if you have `.git/hooks/pre-commit` and `.git/hooks/prepare-commit-msg` files in your repo.

**Comment by RauknRolf** [comment]
I'm sorry, I was too fast sending in this report. 

This is not happening in another repo.
I do not have any commit hooks.

When I tried to commit using another tool on this repo, I experienced similar slow response. It has got to do with my repo rather than Fork. Sorry for the inconvenience. 

Thank you very much for the fantastic tool you are developing!

**Comment by DanPristupov** [comment]
> This is not happening in another repo.

Great!

1. There must be a hook... A commit can not take 30 seconds. Can you double check the content of the `git/hooks` directory?

2. Just to be sure, can you run Repository -> Benchmark and show the report?

**Comment by RauknRolf** [comment]
The benchmark report:
```
Measuring system latency
  Measurements (s): 0,050*, 0,046, 0,045, 0,044, 0,044, 0,046, 0,043*
  Avg: 0,045s

Reading changed files
  Measurements (s): 0,156*, 0,148, 0,152, 0,151, 0,151, 0,154, 0,147*
  Avg: 0,151s

Reading branches and tags
  Measurements (s): 0,172*, 0,171, 0,167, 0,170, 0,166, 0,162, 0,161*
  Avg: 0,167s

Reading commits
  Measurements (s): 0,858*, 0,835, 0,853, 0,856, 0,851, 0,847, 0,825*
  Avg: 0,848s

 Total time: 1,212s
 Avg test time: 0,303s
 Score = 10 / 0,303 = 33,0

0,045
0,151
0,167
0,848
33,0
```

**Comment by RauknRolf** [comment]
You are, of course, right. There is a pre-commit file with this content:

```
#!/bin/sh

# Source the pre-commit scripts. Place this file inside of .git/hooks

if [ ! -d "hooks/pre-commit.d" ]; then
    exit 0
fi

for hook in hooks/pre-commit.d/*.sh; do
    $hook
    exit_code=$?

    if [ $exit_code != 0 ]; then
      exit $exit_code
    fi
done
```
Do I need to remove that file?

**Comment by DanPristupov** [comment]
> Do I need to remove that file?

I'd recommend to not delete, but to rename it. Then investigate when and why that hook appeared in the first place.

**Comment by DanPristupov** [comment]
This pre-commit hooks runs all hooks in the `.git/hooks/pre-commit.d/` directory. So, check what's inside.

https://stackoverflow.com/questions/26624368/handle-multiple-pre-commit-hooks

**Comment by RauknRolf** [comment]
It turns out that my problem was that I had not performed manual housekeeping in a **very** long time. 

I tried `git gc` but it did not seem to be enough.

My problem was similar to what is described in 
https://stackoverflow.com/questions/28633956/why-does-git-keep-telling-me-its-auto-packing-the-repository-in-background-for
and 
https://stackoverflow.com/questions/8633981/what-does-auto-packing-the-repository-for-optimum-performance-mean

When committing from bash, I could see this auto-packing operation kick in and complete with a warning:
```
Auto packing the repository in background for optimum performance.
See "git help gc" for manual housekeeping.
Enumerating objects: 38383, done.
Counting objects: 100% (312/312), done.
Delta compression using up to 8 threads
Compressing objects: 100% (312/312), done.
Writing objects: 100% (312/312), done.
Total 312 (delta 207), reused 0 (delta 0), pack-reused 0
Checking connectivity: 2021253, done.
warning: There are too many unreachable loose objects; run 'git prune' to remove them.

```
finally `git prune` solved the problem. 

About the hooks: The folder `.git/hooks/pre-commit.d/ `did not even exist... I figured it might be stopping on the fact that the folder was missing and eventually timed out, so I created the folder empty. This did not make any difference. Still, I have no idea why this pre-commit file exists at all, but with the problem solved, I will not investigate that further.

Thanks for your help!
